### PR TITLE
`managed_hsm_role_*_ids`: use specific resource id to replace generic nested item id

### PR DESCRIPTION
**Pull request by wuxu92**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

Use specific resource ids for managed hsm role defitnition and role assignment instead of the origin generic id. **It contains a scope field in the middle of the current resource ID, So it's not supported to convert to a New Resource ID Type in this PR**.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_key_vault_managed_hardware_security_module_role_definition` - update resource id with specific definition instead of generic ones
* `azurerm_key_vault_managed_hardware_security_module_role_assignment` - update resource id with specific definition instead of generic ones

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

### CODE REVIEW

**Comment by rcskosir** [comment]
:wave: Thank you for using our new PR template and adding a description! Can you please check the boxes that apply for this PR and also update the changelog portion for this PR? This aids with the review process. Thank you!

File: internal/services/managedhsm/parse/managed_hsm_role_assignment_id.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	Scope        string
+	Name         string
+}
+
+func NewManagedHSMRoleAssignmentID(hsmBaseUrl, scope string, name string) (*ManagedHSMRoleAssignmentId, error) {
+	keyVaultUrl, err := url.Parse(hsmBaseUrl)
+	if err != nil || hsmBaseUrl == "" {
+		return nil, fmt.Errorf("parsing managedHSM nested itemID %q: %+v", hsmBaseUrl, err)
+	}
+	// (@jackofallops) - Log Analytics service adds the port number to the API returns, so we strip it here
```
Comment (mbfrahry) [inline internal/services/managedhsm/parse/managed_hsm_role_assignment_id.go:27]: Is this relevant here? And if so, is it related to the log analytics service?

**Review by mbfrahry — COMMENTED** [review]
Hey @wuxu92 , thanks for this PR! I think it mostly looks good but I do have a question around ports. It looks like we're stripping them but I wasn't under the impression that ports were associated with this id.

If ports do need to be stripped, is there a reason we are only doing it on the role assignment resource and not the role definition resource?

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Thanks for getting that cleaned up but it looks like the unit tests need to be updated

**Review by mbfrahry — APPROVED** [review]
LGTM! Thanks for getting that fixed up @wuxu92

**Pull request merged**